src/NUnitFramework/framework/Internal/Extensions/TestExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// ***********************************************************************
+// Copyright (c) 2021 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using NUnit.Framework.Interfaces;
+
+namespace NUnit.Framework.Internal.Extensions
+{
+    internal static class TestExtensions
+    {
+        public static bool HasLifeCycle(this ITest test, LifeCycle lifeCycle)
```

Comment:
This method has a drawback that it will, perhaps, skip the overriden LifeCycle.SingleInstance fixture within parent LifeCycle.InstancePerTestCase.

Surprisingly, I've added some tests exactly for this case and they pass. Perhaps this is because nested fixture doesn't have wrapping fixture as parent, but I didn't investigate that.

Better solution would be to add something like new LifeCycle.Inherit and make it default one, but that would be a breaking change, so I left it like this for now.